src/client/api/events/sub_outputs_payload.c

```diff
@@ -1,13 +1,36 @@
 // Copyright 2021 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
+#include <string.h>
+
 #include "client/api/events/sub_outputs_payload.h"
 #include "client/api/json_parser/json_utils.h"
 #include "core/address.h"
 #include "core/models/outputs/output_foundry.h"
+#include "core/utils/bech32.h"
 #include "core/utils/iota_str.h"
 #include "core/utils/macros.h"
 
+/**
+ * @brief Validates Bech32 address length
+ *
+ * https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki
+ * the length of Bech32 should be (HRP length) < x <= 90, where the HRP length is 1 to 83 ASCII characters
+ *
+ * @param[in] addr A string of Bech32 address
+ * @return true Valid Bech32 address
+ * @return false Invalid Bech32 address
+ */
+static bool is_valid_bech32_len(char const *const addr) {
```

Comment:
This function looks like it could be used in more places in a code and be publicly available.
One example is in `req_tokens_to_addr_from_faucet` where bech32 validation occurs.

Reply:
right, it could be useful when wallet implementation. moved it to bech32.h